packages/node/src/node.ts

```diff
@@ -102,20 +103,32 @@ export class Node extends Base {
 
       return value.toISOString();
     } else if (Array.isArray(value)) {
-      return value.map((item) => this.sanitizeForEncoding(item));
+      return value.map((item) => this.sanitizeForEncoding(item, maxDepth-1, visitedObjects));
```

Comment:
Arrays are also objects, and although unlikely they could also  be part of a circular reference. You could track them using the `visitedObject` set as well.

Reply:
Thanks, updated